pyface/color.py

```diff
@@ -0,0 +1,260 @@
+
+    Values are rounded to the nearest integer, rather than truncated.
+
+    Parameters
+    ----------
+    channels : iterable of float
+        An iterable of channel values, each value between 0.0 and 1.0,
+        inclusive.
+    maximum : int
+        The maximum value of the integer range.  Common values are 15,
+        65535 or 255, which is the default.
+
+    Returns
+    -------
+    values : tuple of int
+        A tuple of values as integers between 0 and max, inclusive.
+    """
+    return tuple(int(round(channel * maximum)) for channel in channels)
+
+
+def ints_to_channels(values, maximum=255):
+    """ Convert an iterable of integers to floating point channel values.
+
+    Parameters
+    ----------
+    values : tuple of int
+        An iterable of values as integers between 0 and max, inclusive.
+    maximum : int
+        The maximum value of the integer range.  Common values are 15,
+        65535 or 255, which is the default.
+
+    Returns
+    -------
+    channels : iterable of float
+        A tuple of channel values, each value between 0.0 and 1.0,
+        inclusive.
+    """
+    return tuple(value / maximum for value in values)
+
+
+#: A trait holding a single channel value.
+Channel = Range(0.0, 1.0, value=1.0, channel=True)
+
+#: A trait holding three channel values.
+ChannelTuple = Tuple(Channel, Channel, Channel)
+
+#: A trait holding four channel values.
+AlphaChannelTuple = Tuple(Channel, Channel, Channel, Channel)
+
+
+class Color(HasStrictTraits):
+    """ A mutable specification of a color with alpha.
+
+    This is a class designed to be used by user interface elements which
+    need to color some or all of the interface element.  Each color has a
+    number of different representations as channel tuples, each channel
+    holding a value between 0.0 and 1.0, inclusive.  The standard red,
+    green, blue and alpha channels are also provided as convenience
+    properties.
+
+    Methods are provided to convert to and from toolkit-specific color
+    objects.
+
+    Colors implement equality testing, but are not hashable as they are
+    mutable, and so are not suitable for use as dictionary keys.  If you
+    need a dictionary key, use an appropriate channel tuple from the
+    object.
+    """
+
+    #: A tuple holding the red, green, blue, and alpha channels.
+    rgba = AlphaChannelTuple()
+
+    #: A tuple holding the red, green, and blue channels.
+    rgb = Property(ChannelTuple(), depends_on='rgba')
+
+    #: The red channel.
+    red = Property(Channel, depends_on='rgba')
+
+    #: The green channel.
+    green = Property(Channel, depends_on='rgba')
+
+    #: The blue channel.
+    blue = Property(Channel, depends_on='rgba')
+
+    #: The alpha channel.
+    alpha = Property(Channel, depends_on='rgba')
+
+    #: A tuple holding the hue, saturation, value, and alpha channels.
+    hsva = Property(AlphaChannelTuple, depends_on=['rgba'])
```

Comment:
style nit : i'm not sure why `depends_on=['rgba']` here and `depends_on='rgba'` earlier.